### ISSUE TITLE
Chapter 10 - Advantage computed the wrong way?

### ISSUE THREAD

**Issue by JeremieGauthier** [PacktPublishing/Deep-Reinforcement-Learning-Hands-On#74]
https://ai.stackexchange.com/questions/21172/advantage-computed-the-wrong-way

    ipdb> adv_v.shape                                                                                                                            
    torch.Size([128, 128])

    ipdb> vals_ref_v.shape                                                                                                                       
    torch.Size([128])

    ipdb> values_v.detach().shape                                                                                                                
    torch.Size([128, 1])

**Comment by JeremieGauthier** [comment]
I changed the line `adv_v = vals_ref_v - value_v.detach()` to `adv_v = vals_ref_v - value_v.squeeze(-1).detach()`. It seems the convergence is much faster. According to the A2C algorithm, it is just logic to subtract Q(a, s) - V(s) where Q(a, s) and V(s) with the same shape.  

The call to `detach()` is important here as we don't want to propagate the PG into our value approximation head.

**Comment by JeremieGauthier** [comment]
![Selection_028](https://user-images.githubusercontent.com/24579703/82098886-de4bd900-96d3-11ea-9899-2dd6111d12bc.png)

![Selection_027](https://user-images.githubusercontent.com/24579703/82098893-e441ba00-96d3-11ea-9adc-18ab0a6849fc.png)
